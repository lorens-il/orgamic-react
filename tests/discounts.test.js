/**
 * @jest-environment jsdom
 */

 import { render} from "@testing-library/react";

 import Discounts from "../src/components/discounts/Discounts";
 
 import * as RTKQhooks from "../src/api/apiSlice";
 jest.mock("../src/api/apiSlice");
 
 const mockedUseGetDiscountsQuery = jest.spyOn(RTKQhooks, "useGetDiscountsQuery");
 
 describe('Discounts', () => {
 
 it('should create empty Discounts', () => {
 
     mockedUseGetDiscountsQuery.mockReturnValue({data: []});
 
     const component = render(<Discounts/>);
     expect(component).toMatchSnapshot();
 });
 
 it('should create Discounts with list discounts', () => {
 
     mockedUseGetDiscountsQuery.mockReturnValue({data: [{
         "id": "95a3839-cf41-46dc-84bf-63369d3bdf6a",
         "setStyle": 1, 
         "title": "Get 10% off on Vegetables",
         "desc": "Shop our selection of organic fresh vegetables in a discounted price. 10% off on all vegetables.",
         "url": "https://i.ibb.co/SyQPJSB/broccoli.png",
         "alt": "vegetables"
     }]});
 
     const component = render(<Discounts/>);
     expect(component).toMatchSnapshot();
 });
 
 });
 