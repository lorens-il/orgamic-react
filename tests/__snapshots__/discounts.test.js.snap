// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discounts should create Discounts with list discounts 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="discounts"
      >
        <div
          class="container"
        >
          <div
            class="discounts__items"
          >
            <div
              class="discounts__item discounts__item_color-1"
            >
              <div
                class="discounts__wrapper"
              >
                <h2
                  class="discounts__title"
                >
                  Get 10% off on Vegetables
                </h2>
                <div
                  class="discounts__desc"
                >
                  Shop our selection of organic fresh vegetables in a discounted price. 10% off on all vegetables.
                </div>
              </div>
              <div
                class="discounts__picture discounts__picture_option-1"
              >
                <img
                  alt="vegetables"
                  src="https://i.ibb.co/SyQPJSB/broccoli.png"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="discounts"
    >
      <div
        class="container"
      >
        <div
          class="discounts__items"
        >
          <div
            class="discounts__item discounts__item_color-1"
          >
            <div
              class="discounts__wrapper"
            >
              <h2
                class="discounts__title"
              >
                Get 10% off on Vegetables
              </h2>
              <div
                class="discounts__desc"
              >
                Shop our selection of organic fresh vegetables in a discounted price. 10% off on all vegetables.
              </div>
            </div>
            <div
              class="discounts__picture discounts__picture_option-1"
            >
              <img
                alt="vegetables"
                src="https://i.ibb.co/SyQPJSB/broccoli.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Discounts should create empty Discounts 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="discounts"
      >
        <div
          class="container"
        >
          <div
            class="discounts__items"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="discounts"
    >
      <div
        class="container"
      >
        <div
          class="discounts__items"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
